### ISSUE TITLE
HTTP API - cat output has both Content-Length and Transfer-Encoding

### ISSUE THREAD

**Issue by wemeetagain** [ipfs/go-ipfs#1765]
From [RFC2616#4.4](http://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.4):
> Messages MUST NOT include both a Content-Length header field and a non-identity transfer-coding.

The `cat` http response message includes both a content-length header field and a non-identity transfer-encoding.

proof:
```bash
$ echo 'test' > test.txt
$ ipfs add test.txt 
added QmeomffUNfmQy76CQGy9NdmqEnnHU9soCexBnGU3ezPHVH test.txt
$ curl -i http://localhost:5001/api/v0/cat?arg=QmeomffUNfmQy76CQGy9NdmqEnnHU9soCexBnGU3ezPHVH                                            
HTTP/1.1 200 OK
Content-Length: 5
Transfer-Encoding: chunked
X-Stream-Output: 1

test
```
for comparison, the `block get` output:
```bash
$ curl -i http://localhost:5001/api/v0/block/get?arg=QmeomffUNfmQy76CQGy9NdmqEnnHU9soCexBnGU3ezPHVH                                       
HTTP/1.1 200 OK
Transfer-Encoding: chunked
X-Stream-Output: 1



test
```

**Comment by jbenet** [comment]
SGTM. let's
- [ ] add a sharness test.
- [ ] fix the bug.

**Comment by rht** [comment]
Fixed in #2229 and test included.

**Comment by whyrusleeping** [comment]
fixed in #2465
